### ISSUE TITLE
🐛 Factory shift the time when calling ->now()

### ISSUE THREAD

**Issue by sitenzo** [briannesbitt/Carbon#2177]
Hello,

I encountered an issue with the following code:
```php
$Test = new Factory([
    'locale' => 'fr_FR',
    'timezone' => 'Europe/Amsterdam',
]);
echo '$Test   = ' . $Test->now()->format('H:i');
```

Carbon version: **2.39.0**

PHP version: **7.4.8**

I expected to get:

```
Current time from timezone (Europe/Amsterdam)
```

But I actually get:

```
Current UTC time
```
for same result paste undernead code in https://try-carbon.herokuapp.com/

```php
$Test = new Factory([
    'locale' => 'fr_FR',
    'timezone' => 'Europe/Amsterdam',
]);

echo '$Test   = ' . $Test->now()->format('H:i');
echo "\n";

$Test2 = new Carbon();
$Test2->settimezone('Europe/Amsterdam');
echo '$Test 2 = ' . $Test2->format('H:i');
```

Thanks!

**Comment by kylekatarnls** [comment]
Hello,

The factory actually uses `shiftTimezone()` instead of `setTimezone()`. I don't remember the reason for that. I will inspect.

**Comment by kylekatarnls** [comment]
This actually uses `shiftTimezone()` so `$Test->parse('21:54:00')` will actually be 21:54 in the given timezone. So it can't be simply replaced. It has to use shift or set or to automatically fill the timezone parameter before the instance is created.

**Comment by kylekatarnls** [comment]
It will be fixed in 2.40.0 release.
